### ISSUE TITLE
Can't allocate workers to volunteer events

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/GBE2#566]
Here's my process:
- made a new Show in GBE 2016, 2 hours long, booked it Friday night, gave it a cheesy description
- made a new volunteer opportunity in the show, gave it some volunteers, and a time and a length.

Tried the "Allocate Workers" button.

Total failure - Type error.

Example URL:

http://localhost:8282/scheduler/edit/GenericEvent/1296

first part of error:
```

TypeError at /scheduler/edit/GenericEvent/1296

'function' object has no attribute '__getitem__'

Request Method: 	GET
Request URL: 	http://localhost:8282/scheduler/edit/GenericEvent/1296
Django Version: 	1.6.11
Exception Type: 	TypeError
Exception Value: 	

'function' object has no attribute '__getitem__'

Exception Location: 	/vagrant/expo/scheduler/views.py in show_potential_workers, line 569
Python Executable: 	/usr/bin/python
Python Version: 	2.7.6
Python Path: 	

['/vagrant/expo',
 '/home/vagrant/src/cmsplugin-image-gallery-master',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/usr/lib/python2.7/lib-old',
 '/usr/lib/python2.7/lib-dynload',
 '/usr/local/lib/python2.7/dist-packages',
 '/usr/lib/python2.7/dist-packages']

Server time: 	Tue, 17 Nov 2015 02:37:43 +0000
Traceback Switch to copy-and-paste view

    /usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py in get_response

                            response = wrapped_callback(request, *callback_args, **callback_kwargs)

         ...
    ▶ Local vars
    /vagrant/expo/scheduler/views.py in edit_event

                return edit_event_display(request, item)

         ...
    ▶ Local vars
    /vagrant/expo/scheduler/views.py in edit_event_display

                    context.update(show_potential_workers(item))

         ...
    ▶ Local vars
    /vagrant/expo/scheduler/views.py in show_potential_workers

                                            available in eval(v.availability)])]

         ...
    ▶ Local vars 
```

**Comment by bethlakshmi** [comment]
@jonkiparsky - giving this to you because I think you've done way more work here than I have...

**Comment by bethlakshmi** [comment]
This also blocks testing whether or not my list of volunteers would be specifically for GBE 2016, or where I get all volunteers.

The right answer is - the only volunteers it gets should be those that volunteered for this conference.  I realize that this will probably happen anyway, if they volunteered for dates in 2015 that are obviously not there in 2016.  But it is entirely possible that we could create the correct conditions for an overlap, so I'd rather filter event <-> volunteer.

**Comment by jonkiparsky** [comment]
This turns out to depend on GBE-488, which is in progress

**Comment by bethlakshmi** [comment]
Aha!  Sorry for not reading this before responding to email.
Cool.  It had a horrible very bad feeling when I was breaking it. :(
"very bad" as in "the fix to this does not feel easy"

**Comment by jonkiparsky** [comment]
Definitely not easy, but also kind of important. As in, oops, I designed a one-shot system, now I have to fix that. :(

**Comment by jonkiparsky** [comment]
@bethlakshmi Please review, I think this is fixed now.

**Comment by bethlakshmi** [comment]
Woot!  Agree!  This is fixed by #488.  Repeated the steps above and this time I succceed!
